### PR TITLE
add KFP SDK v1 to v2 migration documentation

### PR DESCRIPTION
**Pull request by connor-mccarthy**

### CODE REVIEW

File: content/en/docs/components/pipelines/v2/migration.md [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+</tr>
+</table>
+
+#### VolumeOp and ResourceOp support
+
+**Affects:** KFP OSS users
+
+`VolumeOp` and `ResourceOp` expose direct access to Kubernetes resources within a pipeline definition. There is no support for these features on a non-Kubernetes platforms.
+
+These features will be supported via platform-specific functionality, which includes Kubernetes-specific functionality. This functionality is a work in progress.
```
Comment (chensun) [inline content/en/docs/components/pipelines/v2/migration.md:305]: This section definitely needs a rewrite when we go GA. Should we give more specifics on "work in progress" means ready by 2.0 GA?
Also the SDK syntax for this is ready, up to you to decide when to add the code snippet here.

**Review by chensun — COMMENTED** [review]
nit: It seems the visual could use a bit improvement--no visual divider between the title, and feels a bit disconnected from the code block:
image

/lgtm

Removed approval label so it won't get merged automatically. Up to you when to merge.

File: content/en/docs/components/pipelines/v2/migration.md (hunk 1 above)
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/migration.md:305]: > This section definitely needs a rewrite when we go GA.

For GA we will have full platform-specific docs (both in KFP docs and in kfp-kubernetes library docs), with authoring examples to reference. Is there more you have in mind that I can capture in a TODO?

> Should we give more specifics on "work in progress" means ready by 2.0 GA?

Yes, will update.

> Also the SDK syntax for this is ready, up to you to decide when to add the code snippet here.

I prefer to wait as including may suggest that the BE supports it in the current state, as everything else in the docs is a reflection of the current state. Let me know if you disagree.

**Comment by connor-mccarthy** [comment]
> nit: It seems the visual could use a bit improvement--no visual divider between the title, and feels a bit disconnected from the code block: image
> 
> /lgtm
> 
> Removed approval label so it won't get merged automatically. Up to you when to merge.

@chensun, agreed on the visual improvement. This proved to be pretty challenging to get right due the brittle interaction between Markdown x HTML and the specific Hugo Markdown rendering behavior. I wasn't able to crack it after moderate effort, so prefer to merge and possibly come back to it.

**Review by chensun — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**